backend/de.metas.externalsystem/src/main/java/de/metas/externalsystem/outboundendpoint/ExternalSystemOutboundEndpointRepository.java

```diff
@@ -31,8 +33,18 @@
 @Repository
 public class ExternalSystemOutboundEndpointRepository
 {
+	CCache<ExternalSystemOutboundEndpointId, ExternalSystemOutboundEndpoint> endpointsCache = CCache.<ExternalSystemOutboundEndpointId, ExternalSystemOutboundEndpoint>builder().tableName(I_ExternalSystem_Outbound_Endpoint.Table_Name)
+			.build();
+
 	@NonNull
 	public ExternalSystemOutboundEndpoint getById(@NonNull final ExternalSystemOutboundEndpointId id)
+	{
+		return Check.assumeNotNull(
```

Comment:
* rename getById0 to retrieveById
* make sure `retrieveById` is returning nonnull
* remote this Check.asusmeNotnNull

Reply:
NOTE: AFAIR, if loader method returns null then that result won't be cached anyways...